internal/fetch/lastSuccessfulMeta.go

```diff
@@ -0,0 +1,77 @@
+package fetch
+
+import (
+	"fmt"
+	"github.com/tidwall/gjson"
+	"io/ioutil"
+	"net/http"
+)
+
+type LastSuccessfulMetaRequest struct {
+	SdToken  string
+	SdApiUrl string
+
+	DefaultSdPipelineId int64
+}
+
+func (r *LastSuccessfulMetaRequest) LastSuccessfulMetaURL(jobID int64) string {
+	return fmt.Sprintf("%sjobs/%d/lastSuccessfulMeta", r.SdApiUrl, jobID)
+}
+
+func (r *LastSuccessfulMetaRequest) JobsForPipelineURL(piplineID int64) string {
+	return fmt.Sprintf("%spipelines/%d/jobs", r.SdApiUrl, piplineID)
+}
+
+func (r *LastSuccessfulMetaRequest) JobIdFromJsonByName(json string, jobName string) (int64, error) {
+	result := gjson.Get(json, fmt.Sprintf("#(name==%#v).id", jobName))
+	if !result.Exists() {
+		return 0, fmt.Errorf("jobName %v not found in json", jobName)
+	}
+	return result.Int(), nil
+}
+
+func (r *LastSuccessfulMetaRequest) FetchJobId(roundTripper http.RoundTripper, jobDescription *JobDescription) (int64, error) {
+	if jobDescription.PipelineID == 0 {
+		jobDescription.PipelineID = r.DefaultSdPipelineId
+	}
+	if jobDescription.PipelineID == 0 {
+		return 0, fmt.Errorf("jobDescription does not have pipelineID %#v", jobDescription)
+	}
+	jobsForPipelineURL := r.JobsForPipelineURL(jobDescription.PipelineID)
```

Comment:
You can call `/pipelines/{id}/jobs?jobName={name}`
That will give you an array of just 1 job that matches the name.

Reply:
Ok - will use that.